contracts/strategies/CurveStrategyForLPV1.sol

```diff
@@ -0,0 +1,302 @@
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "CurveStrategyForAv3CRVV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            (uint pendingAvaxRewards, uint pendingCrvRewards) = _claimRewards();
+            uint unclaimedRewards = _estimateRewardConvertedToAvax(pendingAvaxRewards, pendingCrvRewards);
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(pendingAvaxRewards, pendingCrvRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount));
+        _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(amount));
+        totalDeposits = totalDeposits.add(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        uint depositTokenAmount = getDepositTokensForShares(amount);
+        if (depositTokenAmount > 0) {
+            _withdrawDepositTokens(depositTokenAmount);
+            _safeTransfer(address(depositToken), msg.sender, depositTokenAmount);
+            _burn(msg.sender, amount);
+            totalDeposits = totalDeposits.sub(depositTokenAmount);
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private {
+        require(amount > 0, "CurveStrategyForAv3CRVV1::_withdrawDepositTokens");
+        stakingContract.withdraw(amount);
+    }
+
+    function reinvest() external override onlyEOA {
+        (uint pendingAvaxRewards, uint pendingCrvRewards) = _claimRewards();
+        uint unclaimedRewards = _estimateRewardConvertedToAvax(pendingAvaxRewards, pendingCrvRewards);
+        require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "CurveStrategyForAv3CRVV1::reinvest");
+        _reinvest(pendingAvaxRewards, pendingCrvRewards);
+    }
+
+    function _claimRewards() private returns (uint pendingAvaxRewards, uint pendingCrvRewards) {
+        ICurveRewardsClaimer(stakingContract.reward_contract()).get_reward();
+        stakingContract.claim_rewards();
+        uint pendingAvax = IERC20(WAVAX).balanceOf(address(this));
+        uint pendingCrv = IERC20(CRV).balanceOf(address(this));
+        return (pendingAvax, pendingCrv);
+    }
+
+    function _estimateRewardConvertedToAvax(uint pendingAvaxRewards, uint pendingCrvRewards) private view returns(uint) {
+        uint estimatedWAVAX = 0;
+        if (swapPairCrvAvax > address(0)) {
```

Comment:
I really don't know how > works in this case, usually I use !=, just raising in case you see something in regards this but I think it should be fine